File: solutions/python/math-geometry/rotate-image.py
# LeetCode 48 — Rotate Image (Medium)
# Category: Math & Geometry · Approach: Transpose + Reverse
# Time: O(n²) | Space: O(1)
# Source: https://leetcode.com/problems/rotate-image/

def rotate(matrix: list[list[int]]) -> None:
    n = len(matrix)
    for i in range(n):
        for j in range(i + 1, n):
            matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
    for row in matrix:
        row.reverse()

File: solutions/typescript/math-geometry/rotate-image.ts
// LeetCode 48 — Rotate Image (Medium)
// Category: Math & Geometry · Approach: Transpose + Reverse
// Time: O(n²) | Space: O(1)
// Source: https://leetcode.com/problems/rotate-image/

function rotate(matrix: number[][]): void {
  const n = matrix.length;
  for (let i = 0; i < n; i++) {
    for (let j = i + 1; j < n; j++) {
      [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
    }
  }
  for (const row of matrix) row.reverse();
}
